axelor-sale/src/main/java/com/axelor/apps/sale/service/configurator/ConfiguratorServiceImpl.java

```diff
@@ -177,7 +245,34 @@ public void generateProduct(
 
     configurator.setProduct(product);
     product.setConfigurator(configurator);
-    Beans.get(ProductRepository.class).save(product);
+    productRepository.save(product);
+  }
+
+  /**
+   * Private method that fill attr type fields of Class type with json string equivalent of
+   * attrValueMap
+   *
+   * @param <T>
+   * @param attrValueMap
+   * @param type
+   * @param product
+   */
+  private <T extends Model> void fillAttrs(
+      Map<String, Map<String, Object>> attrValueMap, Class<T> type, T product) {
+
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
+    attrValueMap.entrySet().stream()
+        .forEach(
+            attr -> {
+              try {
+                Map<String, Object> fieldValue = attr.getValue();
+                Mapper classMapper = Mapper.of(type);
+                classMapper.set(product, attr.getKey(), mapper.writeValueAsString(fieldValue));
+              } catch (JsonProcessingException e) {
+                TraceBackService.trace(e);
+              }
+            });
```

Comment:
I think it would be better to move this method to the new service and rename it ConfiguratorMapService -> ConfiguratorMetaJsonFieldService